### ISSUE TITLE
RangeError (start): Invalid value: Not in inclusive range 0..176: 179

### ISSUE THREAD

**Issue by NassimMaluli** [domohuhn/mutation-test#22]
What could be causing such an Error ? 

PS C:\Users\M\Desktop\my_app> dart run mutation_test mutation_testing/config.xml --rules mutation_testing/small_test.xml -d
Error while processing:
  RangeError (start): Invalid value: Not in inclusive range 0..176: 179

```

My config.xml : 

<?xml version="1.0" encoding="UTF-8"?>
<mutations version="1.0">
  <!-- License: BSD-3-Clause  -->
  <directories>
    <directory recursive="true">lib
      <matching pattern="\.dart$"/>
    </directory>
  </directories>
  <exclude>
    <!-- lines can also be globally excluded  -->
    <lines begin="1" end="2"/>
  </exclude>
  
  <commands>
   <!-- <command group="test" expected-return="0" working-directory="./">dart //version</command> -->
    
    <command group="test" expected-return="0" working-directory="./">C:\scr\flutter\bin\flutetr.bat test</command>

  </commands>
</mutations>


my small_test.xml : 

<?xml version="1.0" encoding="UTF-8"?>
<mutations version="1.0">
   <!-- License: BSD-3-Clause  -->
   <!-- The rules element describes all mutations done during a mutation test -->
   <!-- The following children are parsed: literal and regex -->
   <!-- A literal element matches the literal text -->
   <!-- A regex element mutates source code if the regular expression matches -->
   <!-- Each of them must have at least one mutation child -->
   <rules>


    <!--this replaces (x op y,x) and (x op y,y)-->
    <regex pattern="([A-Za-z0-9._]+(?:\(\)|\s*))\s+([\*\/\+\-\~])\s+([A-Za-z0-9._]+(?:\(\)|\s*))" dotAll="false">
    <mutation text="$1"/>
    <mutation text="$3"/>
    </regex>
  
  </rules>
  <!-- This element creates a blacklist, allowing you to exclude parts from the mutations -->
  <exclude>

    <!-- excludes anything between two tokens  -->
    <token begin="//" end="\n"/>
    <token begin="#" end="\n"/>
    <token begin="import" end=";"/>

    <!-- excludes anything that matches a pattern  -->
    <regex pattern="/[*].*?[*]/" dotAll="true"/>
    <!-- exclude loops to prevent infinte tests -->
    <regex pattern="[\s]for[\s]*\(.*?\)[\s]*{" dotAll="true"/>
    <regex pattern="[\s]while[\s]*\(.*?\)[\s]*{.*?}" dotAll="true"/>
    <!-- lines can also be globally excluded  -->
    <!-- line index starts at 1  -->
    <lines begin="1" end="2"/>

  </exclude>

</mutations>

```

**Comment by domohuhn** [comment]
There was a bug if the mutation reduced the line length. If this happened at the end of file, then you could get this exception.

It should be fixed in version 1.5.1.

I think your mutation rule can be improved:
([A-Za-z0-9._]+(?:\(\))?)\s*([\*\/\+\-\~])\s*([A-Za-z0-9._]+(?:\(\))?)
This avoids repeating \s multiple times.

**Comment by NassimMaluli** [comment]
Yes , Thank you for the answer and the rule is really helpful.
